CMakeLists.txt

```diff
@@ -368,6 +368,15 @@ if (BUILD_TESTS)
   include(CTest)
   enable_testing()
   message(STATUS "Unit tests are enabled")
+
+  # If CTEST_PARALLEL_LEVEL is not defined, lets define it to ncpus to keep current behavior
+  execute_process(COMMAND "nproc" OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE CORES)
+  if (NOT TEST_JOB_COUNT)
+    set(TEST_JOB_COUNT ${CORES})
+  endif()
```

Comment:
Promoting `TEST_JOB_COUNT` to a CACHE variable will make it show up in CMakeCache.txt and allow IDEs to display a GUI for it:

```cmake
set (TEST_JOB_COUNT "" CACHE STRING "Override number of parallel jobs to use while running tests")
if (NOT TEST_JOB_COUNT)
  execute_process(COMMAND "nproc" OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE TEST_JOB_COUNT)
endif()
```